spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/AggregateApplicationBuilder.java

```diff
@@ -174,20 +135,40 @@ public ProcessorConfigurer via(Class<?> processor) {
 
 		String[] profiles = null;
 
-		ConfigurableApplicationContext parentContext;
-
 		String namespace;
 
 		Class<?> getApp() {
 			return this.app;
 		}
 
-		public void setParentContext(ConfigurableApplicationContext parentContext) {
-			this.parentContext = parentContext;
+		public T as(String... names) {
+			this.names = names;
+			return getConfigurer();
+		}
+
+		public T args(String... args) {
+			this.args = args;
+			return getConfigurer();
+		}
+
+		public T profiles(String... profiles) {
+			this.profiles = profiles;
+			return getConfigurer();
+		}
+
+		public T parent(ConfigurableApplicationContext parentContext) {
```

Comment:
I'd be tempted to move this to the `AggregateApplicationBuilder` itself since it cannot be meaningfully invoked on each child module.